src/Symfony/Component/HttpClient/Response/NativeResponse.php

```diff
@@ -0,0 +1,287 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\Response;
+
+use Symfony\Component\HttpClient\Chunk\FirstChunk;
+use Symfony\Component\HttpClient\Exception\TransportException;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @internal
+ */
+final class NativeResponse implements ResponseInterface
+{
+    use ResponseTrait;
+
+    private $context;
+    private $url;
+    private $resolveRedirect;
+    private $onProgress;
+    private $remaining;
+    private $buffer;
+    private $inflate;
+
+    /**
+     * @internal
+     */
+    public function __construct(\stdClass $multi, $context, string $url, $options, bool $gzipEnabled, array &$info, callable $resolveRedirect, ?callable $onProgress)
+    {
+        $this->multi = $multi;
+        $this->id = (int) $context;
```

Comment:
As this is internal, why not move the int to the typehints? Should result in the same behavior and Symfony can regulate the strictness of this call